docs/examples/00_l5pc_gradient_descent.ipynb

```diff
@@ -0,0 +1,751 @@
+    "\n",
+    "i_delay = 5.0  # ms\n",
+    "i_dur = 90.0  # ms\n",
+    "i_amp = 1.8  # nA\n",
+    "current = jx.step_current(i_delay, i_dur, i_amp, dt, t_max)\n",
+    "cell.soma.branch(0).loc(0.5).stimulate(current)\n",
+    "cell.soma.branch(0).loc(0.5).record()\n",
+    "\n",
+    "cell.set(\"v\", -72.0)\n",
+    "cell.init_states()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6fec6b18",
+   "metadata": {},
+   "source": [
+    "### Generating the observation\n",
+    "\n",
+    "Next, we have to define the observation that we will aim to fit this cell to. In this example, we will fit the cell to a synthetic recording with known ground truth parameters. We will now set these ground truth parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "119ae3b5",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cell.set(\"CaCon_i\", 5e-05)\n",
+    "cell.set(\"CaCon_e\", 2.0)\n",
+    "\n",
+    "cell.apical.set(\"NaTs2T_gNaTs2T\", 0.026145)\n",
+    "cell.apical.set(\"SKv3_1_gSKv3_1\", 0.004226)\n",
+    "cell.apical.set(\"M_gM\", 0.000143)\n",
+    "cell.soma.set(\"NaTs2T_gNaTs2T\", 0.983955)\n",
+    "cell.soma.set(\"SKv3_1_gSKv3_1\", 0.303472)\n",
+    "cell.soma.set(\"SKE2_gSKE2\", 0.008407)\n",
+    "cell.soma.set(\"CaPump_gamma\", 0.000609)\n",
+    "cell.soma.set(\"CaPump_decay\", 210.485291)\n",
+    "cell.soma.set(\"CaHVA_gCaHVA\", 0.000994)\n",
+    "cell.soma.set(\"CaLVA_gCaLVA\", 0.000333)\n",
+    "\n",
+    "cell.axon.set(\"NaTaT_gNaTaT\", 3.137968)\n",
+    "cell.axon.set(\"NapEt2_gNapEt2\", 0.006827)\n",
+    "cell.axon.set(\"KTst_gKTst\", 0.089259)\n",
+    "cell.axon.set(\"CaPump_gamma\", 0.00291)\n",
+    "cell.axon.set(\"CaPump_decay\", 287.19873)\n",
+    "cell.axon.set(\"SKE2_gSKE2\", 0.007104)\n",
+    "cell.axon.set(\"CaHVA_gCaHVA\", 0.00099)\n",
+    "cell.axon.set(\"KPst_gKPst\", 0.973538)\n",
+    "cell.axon.set(\"SKv3_1_gSKv3_1\", 1.021945)\n",
+    "cell.axon.set(\"CaLVA_gCaLVA\", 0.008752)\n",
+    "\n",
+    "# The H-conductance depends on the distance from the soma.\n",
+    "cell.compute_compartment_centers()\n",
+    "direct_dists = distance_direct(cell.soma.branch(0).comp(0), cell)\n",
+    "cell.nodes[\"dist_from_soma\"] = direct_dists\n",
+    "gH_conductance = (-0.8696 + 2.087 * np.exp(cell.basal.nodes[\"dist_from_soma\"] * 0.0031)) * 8e-5\n",
+    "cell.basal.set(\"H_gH\", gH_conductance)\n",
+    "\n",
+    "cell.set(\"Leak_gLeak\", 3e-05)\n",
+    "cell.set(\"Leak_eLeak\", -75.0)\n",
+    "\n",
+    "cell.set(\"eNa\", 50.0)\n",
+    "cell.set(\"eK\", -85.0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "69cd5093",
+   "metadata": {},
+   "source": [
+    "We then run the simulation and save it as the observation $x_o$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "d53b6391",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "x_o = jx.integrate(cell)[0]  # [0] gets rid of the batch-dimension."
```

Comment:
do we want to jit? Probably not much faster at first iteration, but user might want to rerun the cell, so could be nice.